lib/inject.rb

```diff
@@ -26,7 +41,10 @@ def count_equal(argument, elements)
 #   nested_key([:outer, :inner], { outer: { inner: 'value' } })
 #   # => 'value'
 def nested_key(keys, hash)
-  # TODO: implement using inject
+  keys.inject(hash) do |new_hash, key|
+    return nil unless new_hash[key]
+    new_hash[key]
+  end
```

Comment:
This one... I think this works? I'm concerned about the continued return if the value doesn't exist, like:

```
nested_key([:nothing, :fail_sauce], { outer: { inner: 'value' } })
```

I wonder if `.fetch` would be more appropriate here. Still, `#inject` seems like a good choice here.